core/src/jmh/java/com/uber/m3/tally/AbstractReporterBenchmark.java

```diff
@@ -0,0 +1,140 @@
+    );
+
+    private static final String COUNTER_NAME = "counter";
+    private static final String GAUGE_NAME = "gauge";
+    private static final String TIMER_NAME = "timer";
+    private static final String HISTOGRAM_DURATION_NAME = "histogram_duration";
+    private static final String HISTOGRAM_VALUE_NAME = "histogram_value";
+
+    private StatsReporter reporter;
+
+    @Benchmark
+    public void reportCounterBenchmark(IncrementalValueState value) {
+        reporter.reportCounter(COUNTER_NAME, DEFAULT_TAGS, value.incrementAndGet());
+    }
+
+    @Benchmark
+    public void reportGaugeBenchmark(IncrementalValueState value) {
+        reporter.reportCounter(COUNTER_NAME, DEFAULT_TAGS, value.incrementAndGet());
+    }
+
+    @Benchmark
+    public void reportTimerBenchmark(IncrementalValueState value) {
+        reporter.reportTimer(TIMER_NAME, DEFAULT_TAGS, Duration.ofSeconds(value.incrementAndGet()));
+    }
+
+    @Benchmark
+    public void reportHistogramDurationSamplesBenchmark(IncrementalValueState value) {
+        reporter.reportHistogramDurationSamples(
+                HISTOGRAM_DURATION_NAME,
+                DEFAULT_TAGS,
+                DURATION_EXPONENTIAL_BUCKETS,
+                DURATION_EXPONENTIAL_BUCKETS.getDurationLowerBoundFor(10),
+                DURATION_EXPONENTIAL_BUCKETS.getDurationLowerBoundFor(10),
+                value.incrementAndGet()
+        );
+    }
+
+    @Benchmark
+    public void reportHistogramValueSamplesBenchmark(IncrementalValueState value) {
+        reporter.reportHistogramValueSamples(
+                HISTOGRAM_VALUE_NAME,
+                DEFAULT_TAGS,
+                VALUE_EXPONENTIAL_BUCKETS,
+                5.0,
+                5.0,
+                value.incrementAndGet()
+        );
+    }
+
+    @Setup
+    public void setup() {
+        this.reporter = initReporter();
+
+        reporter.reportCounter(COUNTER_NAME, DEFAULT_TAGS, 1);
+        reporter.reportGauge(GAUGE_NAME, DEFAULT_TAGS, 1);
+        reporter.reportTimer(TIMER_NAME, DEFAULT_TAGS, Duration.ofSeconds(1));
+        reporter.reportHistogramValueSamples(
+                HISTOGRAM_VALUE_NAME,
+                DEFAULT_TAGS,
+                VALUE_EXPONENTIAL_BUCKETS,
+                0.1,
+                0.1,
+                1
+        );
+        reporter.reportHistogramDurationSamples(
+                HISTOGRAM_DURATION_NAME,
+                DEFAULT_TAGS,
+                DURATION_EXPONENTIAL_BUCKETS,
+                Duration.ofSeconds(1),
+                Duration.ofSeconds(1),
+                1
+        );
+    }
+
+    @TearDown
+    public void teardown() {
+        reporter.close();
+    }
+
+    public abstract StatsReporter initReporter();
+
+    @State(Scope.Thread)
+    public static class IncrementalValueState {
+        private long x;
+
+        long incrementAndGet() {
+            if (x == Long.MAX_VALUE) {
+                x = 0;
+            }
+            return x++;
```

Comment:
I detect that this code is problematic. According to the [Correctness (CORRECTNESS)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#correctness-correctness), [DLS: Useless increment in return statement (DLS_DEAD_LOCAL_INCREMENT_IN_RETURN)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#dls-useless-increment-in-return-statement-dls-dead-local-increment-in-return).
This statement has a return such as return x++;. A postfix increment/decrement does not impact the value of the expression, so this increment/decrement has no effect. Please verify that this statement does the right thing.

Reply:
```
long val = value.incrementAndGet();
System.out.println(val);
```

> Current value: 321878
Current value: 321879
Current value: 321880
Current value: 321881
Current value: 321882
Current value: 321883
Current value: 321884